metal/compiler.h

```diff
@@ -4,9 +4,11 @@
 #ifndef METAL__COMPILER_H
 #define METAL__COMPILER_H
 
-#define __METAL_DECLARE_VTABLE(type)                        \
-    asm(".weak " #type);                                  \
-    const struct type type                                \
+#define __METAL_DECLARE_VTABLE(type)                      \
+    extern const struct type type;			  
+
+#define __METAL_DEFINE_VTABLE(type)                       \
+    const struct type type                                
```

Comment:
I thought the goal here was to get rid of vtables entirely?

Reply:
Without vtable, we would need to expose each node structure handle/address all the way to the top metal APIs. I was with the impression that we don't want to do that.